dsgrid/registry/project_registry_manager.py

```diff
@@ -302,20 +305,58 @@ def _check_dataset_base_to_project_base_mappings(
         ][
             0
         ]  # this assumes only data_source per dataset
-        dim_table = handler.get_unique_dimension_rows().drop("id")
+        dim_table = (
+            handler.get_unique_dimension_rows().drop("id").drop(DimensionType.DATA_SOURCE.value)
+        )
 
-        associations = project_config.dimension_associations
-        # TODO: check a unified project table against dim_table
-        # project_table = self._make_single_table(project_config, data_source, pivot_dimension)
+        cols = [x.value for x in DimensionType if x not in exclude_dims]
+        project_table = (
+            self._make_single_table(project_config, data_source, pivot_dimension)
+            .select(*cols)
+            .distinct()
+        )
+        result = project_table.intersect(dim_table.select(*cols).distinct())
+        project_table_count = project_table.count()
+        intersect_count = result.count()
+        diff = project_table_count - intersect_count
+        if diff != 0:
+            msg = (
+                f"Dataset {dataset_config.config_id} is missing required dimension records "
+                f"from project {project_config.config_id}: {diff}"
+            )
+            logger.error(msg)
+            self._find_missing_dimension_records(
+                project_config, dataset_config, dim_table, data_source, exclude_dims
+            )
+            raise DSGInvalidDataset(msg)
```

Comment:
This latest commit reduces the time to check dimension records in half, so it is a no-brainer. I left the old code for reporting purposes, but that can be optimized later. This needs review @elainethale  @lixiliu

Reply:
Symmetric_difference equivalent for dataframe:
```
dim_table_unique = dim_table.select(*cols).distinct()
project_table.unionAll(dim_table_unique).except(
    project_table.intersect(dim_table_unique)
)
```